aws/resource_aws_msk_scram_secret.go

```diff
@@ -0,0 +1,216 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/kafka"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAwsMskScramSecret() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsMskScramSecretCreate,
+		Read:   resourceAwsMskScramSecretRead,
+		Update: resourceAwsMskScramSecretUpdate,
+		Delete: resourceAwsMskScramSecretDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"cluster_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"secret_arn_list": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"scram_secrets": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsMskScramSecretCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).kafkaconn
+
+	existingSecrets, err := readMSKClusterSecrets(conn, d.Get("cluster_arn").(string))
+	if err != nil {
+		return fmt.Errorf("failed lookup secrets %s", err)
+	}
+
+	createSecrets := filterMskNewSecrets(expandStringList(d.Get("secret_arn_list").([]interface{})), existingSecrets)
+
+	out, err := associateMSKClusterSecrets(conn, d.Get("cluster_arn").(string), createSecrets)
+	if err != nil {
+		return fmt.Errorf("error associating credentials with MSK cluster: %s", err)
+	}
+	d.SetId(d.Get("cluster_arn").(string))
+
+	if len(out.UnprocessedScramSecrets) != 0 {
+		return fmt.Errorf("there were unprocessed secrets during association: %s", out.UnprocessedScramSecrets)
+	}
+
+	return resourceAwsMskScramSecretRead(d, meta)
+}
+
+func resourceAwsMskScramSecretRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).kafkaconn
+
+	scramSecrets, err := readMSKClusterSecrets(conn, d.Get("cluster_arn").(string))
+	if err != nil {
+		return fmt.Errorf("failed lookup secrets %s", err)
+	}
```

Comment:
Nit: 
```suggestion
	if err != nil {
		return fmt.Errorf("error reading scram secrets for cluster (%s): %w", d.Id(), err)
	}
```

Reply:
we'll probably also need to account for the case that the `MSK Cluster` no longer exists e.g. :
```suggestion
	if err != nil {
	        if !d.IsNewResource() && tfawserr.ErrCodeEquals(err, kafka.ErrCodeNotFoundException) {
	                 log.Printf("[WARN] MSK Cluster (%s) not found, removing from state", d.Id())
			d.SetId("")
			return nil
		}
		return fmt.Errorf("error reading scram secrets for cluster (%s)", d.Id(), err)
	}
```

Reply:
^^ to confirm this behavior we'll need a `Disappears` acctest. let me know if you have any questions or point of reference if you have the time to add this test :)